conduct.md

```diff
@@ -7,16 +7,276 @@
+your team deem it possible, team members may leave the lab, so long as they
+are available via Slack for the allotted lab times. This would be considered
+working remotely, and requires the consent of the other team members or
+approval from the instructor. For further guidance on attendance in labs and
+class please refer to the [course
+website](https://www.gregorykapfhammer.com/teaching/cs203S2019/), contact the
+[instructor](https://www.gregorykapfhammer.com/contact/), or schedule an
+[appointment](https://www.gregorykapfhammer.com/schedule/).
+
+* **Acceptable Attendance (C):**
+  * *Members shall*
+    * Attend all lab sessions on time.
+    * Miss sessions with consent of the course instructor.
+    * Leave labs early with consent of all team members or the course instructor.
+    * Be available on Slack during allotted lab times.
+* **Unacceptable Attendance (U):**
+  * *Members shall not*
+    * Miss a lab session without consent of the course instructor.
+    * Arrive more than 10 minutes late to a lab session.
+    * Leave labs early while still needed by all team members.
+
+## Deadlines
+
+To determine appropriate policies for deadlines, it is first important to
+break up deadlines into the different types of possible deadlines for the
+course. The deadlines categories that are possible are those when working
+directly with group members to finish work necessary to move onto other parts
+of the lab, along with completing your assigned portion of the work by the due
+date of the lab.
+
+* **Acceptable Behavior (C):**
+  * *Members shall*
+    * Talk with team members and discuss work needed to be done.
+    * Set up a reasonable time frame for when the work will be completed
+  leaving time for the rest of the lab.
+    * Take into consideration other people's schedules and activities as it may
+    not align with yours.
+    * If you can not come up with an agreed upon time frame, have a vote
+    between all team members who are present to determine the time frame.
+    * If problems still arise then seek out other team members.
+    * Grade yourself based on the grading scale given to determine how well you
+    followed the deadlines section.
+    * All software project assignments completed have a specified due date,
+    whereas the electronic copy must be uploaded to GitHub through the
+    designated repository before the assigned deadline.
+    * Software projects will be completed according to the deadline set by the
+    students, and within the one week period before the next laboratory session
+    begins.
+* **Unacceptable Behavior (U):**
+  * *Members shall not*
+    * Fail to discuss things with team members on due dates.
+    * Fail to complete work on time.
+    * Fail to participate in votes on material.
+    * Fail to communicate reasons for work being late and not on time.
+    * If work is submitted after the specified deadline, the assignment will not
```

Comment:
Still concerned about this section -- it seems these are all things members *should* do, minus the failing part. Thought should be given on if this should be reorganized.

Additionally, `Fail to discuss things with team members on due dates` is unclear -- do you discuss on the day it is due, or discuss the due date itself, or what?

Reply:
@aubreypc I just made a pr for this issue. https://github.com/aubreypc/cs203-S2019-conduct/pull/45 Do you think it is ok to be merged?
Btw, @Michionlion. I feel `Fail to participate in votes on the material.` should just be moved or changed to `Fail to participate in votes on the material without reason.` What is your opinion since you pointed out this section?

Reply:
@Michionlion @Lancasterwu  Personally, I think it's good that this PR states them explicitly as negations of the guidelines since that clarifies that each guideline is not just a suggestion, but a rule. Rewording many of these could introduce ambiguity --- e.g. "shall not complete work after the deadline" is a rule about _completing work_, whereas "shall not fail to complete work on time" is a rule about _following deadlines_.

Reply:
Ultimately this is up to you guys (and I've already approved these), but I still think it is confusing.

Reply:
@aubreypc I agree with your comment and I think it is better to keep in the "fail to" statements. Just made some changes